### PR TITLE
Add option to round up CPU quota

### PR DESCRIPTION
**Pull request by wallee94**
Add a `RoundQuotaFunc` option to control how the CPU quota should be rounded. The default is still rounding down (floor).
This is related to issue https://github.com/uber-go/automaxprocs/issues/78

### CODE REVIEW

**Comment by wallee94** [comment]
I just noticed there was an open PR already. This one's also related to the discussion in https://github.com/uber-go/automaxprocs/pull/68

File: maxprocs/maxprocs.go
Comment (abhinav) [inline maxprocs/maxprocs.go:76]: Not a maintainer, but thinking out loud:
Should this maybe be an enum "Rounding"?

```
type Rounding int

const (
  RoundDown Rounding = iota
  RoundUp
)

func RoundQuota(Rounding) Option
```

**Review by abhinav — COMMENTED** [review]
Thanks for making this change, @wallee94!
I think the maintainers might find this change more digestible. 
I've left a comment about one possible improvement, but I'll defer to the maintainers on it.

CC @r-hang @sywhang @mway

File: maxprocs/maxprocs.go
Comment (sandyskies) [reply maxprocs/maxprocs.go:76]: how about ceil and floor？

Comment (wallee94) [reply maxprocs/maxprocs.go:76]: Thanks @abhinav! Sounds good. I'm taking some time tomorrow to push that change and fix the codecov check

Comment (wallee94) [reply maxprocs/maxprocs.go]: Added. I had to add it in runtime to avoid having circular dependencies with maxprocs

**Comment by wallee94** [comment]
Added a few tests for the new lines in runtime. I don't have linux so I couldn't run them on my local. I'll have to wait for the checks here 👀

**Comment by sandyskies** [comment]
@sywhang

File: internal/runtime/cpu_quota_linux.go
Comment (abhinav) [inline internal/runtime/cpu_quota_linux.go:50]: readability suggestion: this might be an if-else situation over `x := default; if { x = override }` since both involve computation.

```suggestion
	var maxProcs int
	if roundOpt == Ceil {
		maxProcs = int(math.Ceil(quota))
	} else {
		maxProcs = int(math.Floor(quota))
	}
```

File: internal/runtime/runtime.go
Comment (abhinav) [inline internal/runtime/runtime.go:35]:
```suggestion
// Rounding controls how the CPU quota value should be rounded to an int.
```

File: internal/runtime/runtime.go
Comment (abhinav) [inline internal/runtime/runtime.go:41]:
```suggestion
	// Ceil is used to return a CPU quota rounded up.
	Ceil Rounding = iota
	// Floor is used to return a CPU quota rounded down.
```

File: maxprocs/maxprocs.go
Comment (abhinav) [inline maxprocs/maxprocs.go:76]: With the argument being an `internal` type, it'll be impossible for a caller to actually specify this.

We need to declare an analogous type out here.

```suggestion
type Rounding = iruntime.Rounding

const (
	Ceil  Rounding = iruntime.Ceil
	Floor Rounding = iruntime.Floor
)

// RoundQuota controls whether the CPU quota should be rounded using ceil or floor.
func RoundQuota(v Rounding) Option {
```

(Ideally with the documentation copied over.)

**Review by abhinav — COMMENTED** [review]
LGTM save minor edits and a suggested fix for being able to use the RoundQuota option.
I'm not a maintainer, though, so I'll defer to @sywhang @r-hang and @mway.

File: internal/runtime/runtime.go
Comment (wallee94) [reply internal/runtime/runtime.go]: Thanks! I ran the linter this time

File: maxprocs/maxprocs.go
Comment (wallee94) [reply maxprocs/maxprocs.go]: Also, TIL that packages in the internal folder aren't exported 😅

**Comment by wallee94** [comment]
I finally set up a linux machine to run those tests! I also changed the implementation to take a round function in the option instead of setting the operation with a const value. I felt it was simpler to use and maintain.

**Review by r-hang — APPROVED** [review]
looks good to me.

Does it make sense to test the round == nil case within internal/runtime since that's newly added?

**Comment by wallee94** [comment]
@r-hang Sure! Added a test to check that the value is rounded down then the round arg is nil

**Pull request merged**
